src/App.tsx

```diff
@@ -1,223 +1,86 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
 
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { ErrorNotification } from
+  './components/ErrorNotification/ErrorNotification';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { FilterType } from './types/FiltersType';
+
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState(FilterType.all);
+  const [errorText, setErrorText] = useState('');
 
   useEffect(() => {
     // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    if (user) {
+      getTodos(user.id)
+        .then(setTodos)
+        .catch(() => setErrorText('Unable to load a todos'));
+    }
   }, []);
 
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (filterType) {
+        case FilterType.active:
+          return !todo.completed;
+
+        case FilterType.completed:
+          return todo.completed;
+
+        default:
+          return todo;
+      }
+    });
```

Comment:
better to move filter function somewhere

Reply:
done